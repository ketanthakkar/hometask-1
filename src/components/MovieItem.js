import React from 'react';
import Image from './Image';
import '../styleModules/MovieItem.css';

const MovieItem = ({url, name, date, genre}) => {
    return (
        <div className = 'movieItem-wrapper'>
            <Image src = {url} alt = {name}/>
            <span>{name}</span><br/>
            <span>{date}</span><br/>
            <span>{genre.join(', ')}</span>
        </div>
    );
}

export default MovieItem;
