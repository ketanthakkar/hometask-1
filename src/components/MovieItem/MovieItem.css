.movieItem-wrapper {
    width:25%;
    float: left;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items:center;
    margin-top:20px;
}