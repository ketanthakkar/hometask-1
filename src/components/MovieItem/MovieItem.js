import React from 'react';
import Image from '../Image';
import './MovieItem.css';

import { Link } from 'react-router-dom';

const MovieItem = ({url, name, date, genre, id}) => {
    return (
        <Link to ={ `/search/${id}`}>
        <div className = 'movieItem-wrapper'>
            <Image src = {url} alt = {name}/>
            <span>{name}</span><br/>
            <span>{date}</span><br/>
            <span>{genre.join(', ')}</span>
        </div>
        </Link>
    );
}

export default MovieItem;
