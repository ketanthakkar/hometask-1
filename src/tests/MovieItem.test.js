import React from 'react';
import { shallow } from 'enzyme';
import MovieItem from '../components/MovieItem';

describe('MovieItem', () => {
  it('shoud render MovieItem component', () => {
    const render = shallow(
      <MovieItem 
        url = 'url' 
        name = 'name' 
        date = 'date' 
        genre = {['genre']}
      />
    );
    expect(render.find('div').hasClass('movieItem-wrapper')).toBe(true);
    expect(render).toMatchSnapshot();
  })

  it('should not modify props', () => {
    const  url = 'url', name = 'name', date = 'date', genre = ['genre'];
    const render = shallow(
      <MovieItem 
      url = {url} 
      name = {name} 
      date = {date} 
      genre = {genre}
    />
    );
    expect(url).toEqual('url');
    expect(name).toEqual('name');
    expect(date).toEqual('date');
    expect(genre[0]).toEqual('genre');
    expect(render).toMatchSnapshot();
  })
})