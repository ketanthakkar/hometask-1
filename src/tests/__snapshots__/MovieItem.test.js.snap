// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieItem shoud render MovieItem component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieItem
    date="date"
    genre={
      Array [
        "genre",
      ]
    }
    name="name"
    url="url"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Image
          alt="name"
          src="url"
        />,
        <span>
          name
        </span>,
        <br />,
        <span>
          date
        </span>,
        <br />,
        <span>
          genre
        </span>,
      ],
      "className": "movieItem-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "name",
          "src": "url",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "name",
        },
        "ref": null,
        "rendered": "name",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "date",
        },
        "ref": null,
        "rendered": "date",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "genre",
        },
        "ref": null,
        "rendered": "genre",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Image
            alt="name"
            src="url"
          />,
          <span>
            name
          </span>,
          <br />,
          <span>
            date
          </span>,
          <br />,
          <span>
            genre
          </span>,
        ],
        "className": "movieItem-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "name",
            "src": "url",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "name",
          },
          "ref": null,
          "rendered": "name",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "date",
          },
          "ref": null,
          "rendered": "date",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "genre",
          },
          "ref": null,
          "rendered": "genre",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MovieItem should not modify props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieItem
    date="date"
    genre={
      Array [
        "genre",
      ]
    }
    name="name"
    url="url"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Image
          alt="name"
          src="url"
        />,
        <span>
          name
        </span>,
        <br />,
        <span>
          date
        </span>,
        <br />,
        <span>
          genre
        </span>,
      ],
      "className": "movieItem-wrapper",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "name",
          "src": "url",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "name",
        },
        "ref": null,
        "rendered": "name",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "date",
        },
        "ref": null,
        "rendered": "date",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "genre",
        },
        "ref": null,
        "rendered": "genre",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Image
            alt="name"
            src="url"
          />,
          <span>
            name
          </span>,
          <br />,
          <span>
            date
          </span>,
          <br />,
          <span>
            genre
          </span>,
        ],
        "className": "movieItem-wrapper",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "name",
            "src": "url",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "name",
          },
          "ref": null,
          "rendered": "name",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "date",
          },
          "ref": null,
          "rendered": "date",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "genre",
          },
          "ref": null,
          "rendered": "genre",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
